packages/sync-actions/test/types-sync-fields.spec.js

```diff
@@ -336,4 +336,28 @@ describe('Actions', () => {
       ])
     })
   })
+  describe('should ignore required field in fieldDefinition', () => {
+    beforeEach(() => {
+      before = createTestType({
+        fieldDefinitions: [
+          {
+            name: 'first',
+            required: true,
+          },
+        ],
+      })
+      now = createTestType({
+        fieldDefinitions: [
+          {
+            name: 'first',
+            required: false,
```

Comment:
Sorry, I am not following: we change the value but don't create an update action. Can you elaborate?

Reply:
totally agree the test is already confusing 👍 (gonna rename it better and add more context to test), there is some fields that is immutable or unchangeable so there is no `update actions` for it, which apply for the this field `FieldDefinition->required` (maybe also that should be mentioned in the public doc cause there is no clue cause it also apply for `resourceTypeIds`, `FieldDefinition->Name`).

and the current behaviour is to just ignore it for example in product entity , `[productType](https://docs.commercetools.com/api/projects/products#product)`  field `Cannot be changed.`

for above field `required`, changing it causing throwing errors 😞, and thought of adding test for it.

already been thinking about adding util for that case and pass it as config option, so it doesnt require major upgrade.

does it make sense?

Reply:
Yeah thanks makes sense.